src/lib/containers/widgets/ImageButtonWithTooltip.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { FunctionComponent } from 'react'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import ReactTooltip from 'react-tooltip'
+import { TOOLTIP_DELAY_SHOW } from '../table/SynapseTableConstants'
+import { IconDefinition } from '@fortawesome/fontawesome-svg-core'
+import { Dropdown } from 'react-bootstrap'
+
+type CustomImageProps = {
+  svgImg: string
+  altText: string
+}
+
+type ImageButtonWithTooltipProps = {
+  image: IconDefinition | CustomImageProps
+  callbackFn: Function
+  tooltipText: string
+  cssClass?: string
+  idForToolTip: string
+  isDropdownToggle?: boolean
+}
+
+export const ImageButtonWithTooltip: FunctionComponent<ImageButtonWithTooltipProps> = ({
+  image,
+  idForToolTip,
+  callbackFn,
+  tooltipText,
+  cssClass,
+  isDropdownToggle,
+}) => {
+  var buttonImage = <></>
+  if ('svgImg' in image) {
```

Comment:
I think we've generally used the syntax `if(image.svgImage) { }`, is there a reason to prefer using the `in` keyword?

Reply:
Yes, there is a subtle difference if how those work
`image` type can be either `CustomImageProps` or `IconDefinition` in which case it will not have the `svgImg` prop so `image.svgImage` would result in TS screaming about that.
the `in` operator tests if the  property is in the object (or its' prototype chain) 
for example in 
`const _someobj = { prop1: undefined, prop2: 'something else'}`
`('prop1' in _someobj)` will evaluate to true even if `_someobj.prop1` is falsey
So by using `in` operator I'm checking the type of the object rather than the value of the property

Reply:
gotcha, this is good to know!